src/util/sync.js

```diff
@@ -64,7 +64,6 @@ export const sync = (thoughtIndexUpdates = {}, contextIndexUpdates = {}, { local
     : []
 
   return Promise.all(localPromises).then(() => {
-
     // firebase
     if (remote) {
       return syncRemote(thoughtIndexUpdates, contextIndexUpdates, recentlyEdited, updates, callback)
```

Comment:
I'm getting the following error when trying to sync to Firebase:

```js
Reference.update failed: First argument contains an invalid key (3/11) in property 
'users.m9S244ovF7fVrwpAoqoWxcz08s52.recentlyEdited.📒 Journal.2020:'. Keys must be non-empty strings 
and can't contain ".", "#", "$", "/", "[", or "]"
```

We need to `firebaseEncode` and `firebaseDecode` the keys in the tree in order to save it to Firebase. This isn't needed with `thoughtIndex` and `contextIndex` since the keys are already hashed.

Option 1: Stringify the entire `recentlyEdited` tree and save the whole tree to Firebase on each edit (inefficient)
Option 2: Encode they keys and only update the changed nodes in the tree on edit.

You can test this out by logging in via the Log In link in the footer and trying to save a thought with an invalid character such as "3/11".

Reply:
Agreed! Option 1 will be very inefficient. Let's encode provided `oldContext` and `newContext` for all the `recentlyEditedTree` methods. All the nodes will have encoded keys . We won't even have to decode the keys of tree Object to render recently edited list because `path` property stored by leaf nodes won't be encoded.